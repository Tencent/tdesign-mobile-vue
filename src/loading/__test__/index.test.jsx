import { mount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';
import Loading from '../loading';
import Spinner from '../icon/spinner';
import { nextTick } from 'vue';

const sleep = (duration) =>
  new Promise((resolve) =>
    setTimeout(() => {
      resolve();
    }, duration),
  );


describe('Loading.vue', () => {
  describe('props', () => {
    it(':delay', () => {
      const wrapper = mount(() => <Loading delay={1000}></Loading>);
      const loading = wrapper.find('.t-loading');
      setTimeout(() => {
        expect(loading.exists()).toBeTruthy();
      }, 1000);
    });

    it(':inheritColor', () => {
      const wrapper = mount(Loading, {
        props: {
          inheritColor: true,
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':pause', () => {
      const wrapper = mount(Loading, {
        props: {
          pause: true,
        },
      });
      expect(wrapper.element).toMatchSnapshot();
      const iconWrapper = wrapper.find('.t-loading__gradient');
      const iconWrapperStyle = iconWrapper.attributes().style;
      expect(iconWrapperStyle).toContain('animation-play-state: paused');
    });

    it(':reverse', () => {
      const wrapper = mount(() => <Loading reverse={true} />);
      expect(wrapper.element).toMatchSnapshot();
      const iconWrapper = wrapper.find('.t-loading__gradient');
      const iconWrapperStyle = iconWrapper.attributes().style;
      expect(iconWrapperStyle).toContain('animation-direction: reverse');
    });

    it(':theme', () => {
      const wrapper = mount(() => <Loading theme={'spinner'} />);
      const icon = wrapper.findComponent(Spinner);
      expect(icon.exists()).toBeTruthy();
    });
  });

  describe('slots', () => {
    it(':text', () => {
      const customLoading = 'custom-loading-text';
      const wrapper = mount(() => (
        <Loading>{{ text: () => <p className={customLoading}>{customLoading}</p> }}</Loading>
      ));
      const text = wrapper.find('.t-loading__text');
      expect(text.exists()).toBeTruthy();
      expect(text.element.innerHTML).toBe(`<p class="${customLoading}">${customLoading}</p>`);
    });
  });
});

describe('spinner.vue', () => {
  describe('render', () => {
    it(':text', () => {
      const wrapper = mount(() => <Spinner />);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
