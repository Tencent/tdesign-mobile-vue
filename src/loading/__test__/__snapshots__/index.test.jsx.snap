// Vitest Snapshot v1

exports[`Loading.vue > props > :inheritColor 1`] = `
<div
  class="t-loading"
  style="font-size: 20px;"
>
  <svg
    class="t-loading__gradient t-icon-loading"
    height="1em"
    style="animation-duration: 800ms; width: 20px; height: 20px;"
    version="1.1"
    viewBox="0 0 12 12"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <foreignobject
      height="12"
      width="12"
      x="0"
      y="0"
    >
      <div
        class="t-loading__gradient-conic"
      />
    </foreignobject>
  </svg>
  <!---->
  <!---->
</div>
`;

exports[`Loading.vue > props > :pause 1`] = `
<div
  class="t-loading"
  style="font-size: 20px;"
>
  <svg
    class="t-loading__gradient t-icon-loading"
    height="1em"
    style="animation-play-state: paused; animation-duration: 800ms; width: 20px; height: 20px;"
    version="1.1"
    viewBox="0 0 12 12"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <foreignobject
      height="12"
      width="12"
      x="0"
      y="0"
    >
      <div
        class="t-loading__gradient-conic"
      />
    </foreignobject>
  </svg>
  <!---->
  <!---->
</div>
`;

exports[`Loading.vue > props > :reverse 1`] = `
<div
  class="t-loading"
  style="font-size: 20px;"
>
  <svg
    class="t-loading__gradient t-icon-loading"
    height="1em"
    style="animation-direction: reverse; animation-duration: 800ms; width: 20px; height: 20px;"
    version="1.1"
    viewBox="0 0 12 12"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <foreignobject
      height="12"
      width="12"
      x="0"
      y="0"
    >
      <div
        class="t-loading__gradient-conic"
      />
    </foreignobject>
  </svg>
  <!---->
  <!---->
</div>
`;

exports[`spinner.vue > render > :text 1`] = `
<span
  class="t-loading__spinner"
>
  
  <i
    class="t-loading__spinner--line t-loading__spinner--line-1"
  />
  <i
    class="t-loading__spinner--line t-loading__spinner--line-2"
  />
  <i
    class="t-loading__spinner--line t-loading__spinner--line-3"
  />
  <i
    class="t-loading__spinner--line t-loading__spinner--line-4"
  />
  <i
    class="t-loading__spinner--line t-loading__spinner--line-5"
  />
  <i
    class="t-loading__spinner--line t-loading__spinner--line-6"
  />
  <i
    class="t-loading__spinner--line t-loading__spinner--line-7"
  />
  <i
    class="t-loading__spinner--line t-loading__spinner--line-8"
  />
  <i
    class="t-loading__spinner--line t-loading__spinner--line-9"
  />
  <i
    class="t-loading__spinner--line t-loading__spinner--line-10"
  />
  <i
    class="t-loading__spinner--line t-loading__spinner--line-11"
  />
  <i
    class="t-loading__spinner--line t-loading__spinner--line-12"
  />
  
</span>
`;
